phase1/vsphere/vSphere.jsonnet

```diff
@@ -102,8 +102,14 @@ function(config)
             memory: cfg.vSphere.memory,
             enable_disk_uuid: true,
             datacenter: cfg.vSphere.datacenter,
-            resource_pool: cfg.vSphere.resourcepool,
-	    skip_customization: true,
+            skip_customization: true,
+            resource_pool: (
+              if cfg.vSphere.useresourcepool == "no" then
+                if cfg.vSphere.placement == "host" then cfg.vSphere.host
+                else cfg.vSphere.cluster
+              else
+                if cfg.vSphere.placement == "host" then std.join("/", ["/", cfg.vSphere.datacenter, "host", cfg.vSphere.host, "Resources", cfg.vSphere.resourcepool])
+                else std.join("/", ["/", cfg.vSphere.datacenter, "host", cfg.vSphere.cluster, "Resources", cfg.vSphere.resourcepool])),
```

Comment:
The resource pool need not be a direct resource pool under host or cluster. It can be a child resource pool of some parent resource pool. For ex. of the form - /vcqaDC/host/vsan-cluster/Resources/rs1/rs2/rs3.

rs3 is a child of rs2 whereas rs2 is a child of rs1 which is child of cluster resource pool. 

Asking user to specify this format is good.